dev/TabView/TabViewItem.cpp

```diff
@@ -489,6 +489,35 @@ void TabViewItem::OnPointerCaptureLost(winrt::PointerRoutedEventArgs const& args
     RestoreLeftAdjacentTabSeparatorVisibility();
 }
 
+// Note that the ItemsView will handle the left and right arrow keys if we don't do so before it does,
+// so this needs to be handled below the items view. That's why we can't put this in TabView's OnKeyDown.
+void TabViewItem::OnKeyDown(winrt::KeyRoutedEventArgs const& args)
+{
+    if (!args.Handled())
+    {
+        // Alt+Shift+Arrow reorders tabs, so we don't want to handle that combination.
+        auto isAltDown = (winrt::Window::Current().CoreWindow().GetKeyState(winrt::VirtualKey::Menu) & winrt::CoreVirtualKeyStates::Down) == winrt::CoreVirtualKeyStates::Down;
+        auto isShiftDown = (winrt::Window::Current().CoreWindow().GetKeyState(winrt::VirtualKey::Shift) & winrt::CoreVirtualKeyStates::Down) == winrt::CoreVirtualKeyStates::Down;
+
+        if (!isAltDown || !isShiftDown)
+        {
+            if (args.Key() == winrt::VirtualKey::Right)
+            {
+                args.Handled(winrt::get_self<TabView>(GetParentTabView())->MoveFocus(FlowDirection() == winrt::FlowDirection::LeftToRight));
```

Comment:
>FlowDirection() == winrt::FlowDirection::LeftToRight [](http://example.com/codeflow?start=85&length=52)

It would be a little clearer to pull this expression out a separate 'moveForward' bool.

Reply:
Good call.